src/Illuminate/Collections/Collection.php

```diff
@@ -1050,6 +1052,37 @@ public function splitIn($numberOfGroups)
         return $this->chunk(ceil($this->count() / $numberOfGroups));
     }
 
+    /**
+     * Get the first item in the collection, but only if exactly
+     * item exists. Otherwise, throw an exception.
+     *
+     * @param  mixed  $key
+     * @param  mixed  $operator
+     * @param  mixed  $value
+     * @return mixed
+     *
+     * @throws \Illuminate\Collections\ItemNotFoundException
+     * @throws \Illuminate\Collections\MultipleItemsFoundException
+     */
+    public function sole($key = null, $operator = null, $value = null)
+    {
+        if (func_num_args() <= 1) {
+            $items = $this->when($key)->filter($key);
+
+            if ($items->isEmpty()) {
+                throw new ItemNotFoundException;
+            }
+
+            if ($items->count() > 1) {
+                throw new MultipleItemsFoundException;
+            }
+
+            return $items->first();
+        }
+
+        return $this->sole($this->operatorForWhere(...func_get_args()));
+    }
```

Comment:
A ternary would be a little clearer, and allows dedenting the whole method:

```suggestion
    public function sole($key = null, $operator = null, $value = null)
    {
        $filter = func_num_args() > 1
            ? $this->operatorForWhere(...func_get_args())
            : $key;

        $items = $this->when($filter)->filter($filter);

        if ($items->isEmpty()) {
            throw new ItemNotFoundException;
        }

        if ($items->count() > 1) {
            throw new MultipleItemsFoundException;
        }

        return $items->first();
    }
```